### ISSUE TITLE
Range picker allows picking date ranges between disabled dates

### ISSUE THREAD

**Issue by arunavdev89** [MariaMelnik/flutter_date_pickers#15]
I have the `RangePicker` widget created using the constructor as follows:

```
  void _onSelectedDateChanged(DatePeriod newPeriod) {
    setState(() {
      _selectedPeriod = newPeriod;
    });
  }
  bool isSelectable(DateTime date) {
    List<DateTime> eventsDates = [
     
      DateTime.now().add(Duration(days: 3)),
      DateTime.now().add(Duration(days: 4)),
      DateTime.now().add(Duration(days: 5)),
      DateTime.now().add(Duration(days: 6))
    ];
    return !eventsDates.any((eventDate) {
      int diffDays = eventDate.difference(date).inDays;
      return (diffDays == 0);
    });
  }
RangePicker(
    selectableDayPredicate: isSelectable,
    selectedPeriod: _selectedPeriod,
    onChanged: _onSelectedDateChanged,
    firstDate: DateTime.now(),
    lastDate: DateTime.now().add(Duration(days: 60)),
    datePickerStyles: styles,
),
```

It allows me to pick a range where the start of the range is before a block of dates that is not selectable and end of the range after the same block of not selectable date. 

See below:

<img width="275" alt="Screen Shot 2020-01-11 at 7 58 51 PM" src="https://user-images.githubusercontent.com/5005018/72212597-d45b6480-34ac-11ea-8085-0974981f8791.png">

I figure I could do the some processing inside onChanged callback to check if the start and end falls between the not selectable dates but it would be nice if the range picker took care of it.

**Comment by MariaMelnik** [comment]
@arunavdev89 thank you for report! It will be fixed soon in next release.

**Comment by Nieruht** [comment]
How can i get Year Picker !Please let me know ! Thank You Brother .

**Comment by MariaMelnik** [comment]
Fixed in 0.1.0
